scripts/build.sh

```diff
@@ -0,0 +1,293 @@
+#!/bin/bash
+
+set -e
+
+BINARY=sonobuoy
+TARGET=sonobuoy
+GOTARGET=github.com/vmware-tanzu/$TARGET
+GOPATH=$(go env GOPATH)
+REGISTRY=sonobuoy
+#REGISTRY=schnake
+IMAGE=$REGISTRY/$TARGET
+LINUX_ARCH=(amd64 arm64)
+WIN_ARCH=amd64
+KIND_CLUSTER=kind
+
+# Not used for pushing images, just for local building on other GOOS. Defaults to
+# grabbing from the local go env but can be set manually to avoid that requirement.
+HOST_GOOS=$(go env GOOS)
+HOST_GOARCH=$(go env GOARCH)
+
+# --tags allows detecting non-annotated tags as well as annotated ones
+GIT_VERSION=$(git describe --always --dirty --tags)
+IMAGE_VERSION=$(git describe --always --dirty --tags)
+IMAGE_TAG=$(echo $IMAGE_VERSION | cut -d. -f1,2)
+IMAGE_BRANCH=$(git rev-parse --abbrev-ref HEAD | sed 's/\///g')
+GIT_REF_SHORT=$(git rev-parse --short=8 --verify HEAD)
+GIT_REF_LONG=$(git rev-parse --verify HEAD)
+
+BUILDMNT=/go/src/$GOTARGET
+BUILD_IMAGE=golang:1.16
+AMD_IMAGE=gcr.io/distroless/static:nonroot
+ARM_IMAGE=gcr.io/distroless/static:nonroot-arm64
+WIN_IMAGE=mcr.microsoft.com/windows/servercore:1809
+TEST_IMAGE="sonobuoy/testimage:v0.1"
+
+unit_local() {
+	GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/cmd/... $GOTARGET/pkg/...
+}
+
+unit() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+stress() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/test/stress/..."
+}
+
+integration() {
+	docker run --rm \
+        -v $(pwd):$BUILDMNT \
+        -v /tmp/artifacts:/tmp/artifacts \
+        -v ${HOME}/.kube/config:/root/.kube/kubeconfig \
+        --env KUBECONFIG=/root/.kube/kubeconfig \
+        -w $BUILDMNT \
+        --env ARTIFACTS_DIR=/tmp/artifacts \
+        --env SONOBUOY_CLI=$SONOBUOY_CLI \
+        --env GODEBUG=x509ignoreCN=0 \
+        --network host \
+        $BUILD_IMAGE \
+    go test ${VERBOSE:+-v} -timeout 3m -tags=integration $GOTARGET/test/integration/...
+}
+
+lint() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "golint -set_exit_status ${VERBOSE:+-v} -timeout 60s $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+vet() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "CGO_ENABLED=0 go vet ${VERBOSE:+-v} -timeout 60s $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+pre() {
+	wget https://github.com/estesp/manifest-tool/releases/download/v1.0.1/manifest-tool-linux-amd64 \
+        -O manifest-tool
+	chmod +x ./manifest-tool
+	echo $DOCKERHUB_TOKEN | docker login --username sonobuoybot --password-stdin
+}
+
+# TODO: Make it easy to build single container for a specific arch
```

Comment:
Do we care about this? Would be relatively trivial to do.

Reply:
I think I added this comment. I found the delta between building 2 containers and 1 container to be quite significant when I was trying to test locally. Would love to see it happen but also it seems to be a pretty rare case.